File: admin_scripts/remove_11.js
// Move the content from the mysterious "11" keys from mutliple projects, and place in the main body of the project files

var fs = require('fs'),
    path = require('path'),
    _ = require('lodash');

ROOT = '../_ssc_projects/';
ELEVEN_REGEX = /---\n\"11\":\s\"?(.*)\"?$/m;
DEFAULT_TEXT_REGEX = /---\n{2,4}## A New Post\n\nEnter text in \[Markdown]\(http:\/\/daringfireball\.net\/projects\/markdown\/\)\. Use the toolbar above, or click the \*\*\?\*\* button for formatting help\./gm;

function project_files(){
  return _.reject(fs.readdirSync(ROOT), function(entry){
    return entry === '.DS_Store';
  });
}

function broken_files(){
  return _.select(project_files(), function(file){
    var content = project_content_from(file);
    return has_eleven(content) && has_default_content(content);
  })
}

function project_content_from(file){
  var path = ROOT + file;
  return fs.readFileSync(ROOT + file, 'utf8');
}

function fixed_content(file){
  // Only receiving projects with '11' and unchanged default body content
  
  // var data = project_data_from(file);
  // data.body = data.attributes['11'];
  // delete data.attributes['11'];
  // return data;
}

function is_broken(content){
  return has_eleven(content) && has_default_content(content);
}

function has_eleven(content) {
  return (ELEVEN_REGEX.exec(content) !== null);
}

function has_default_content(content) {
  return (DEFAULT_TEXT_REGEX.exec(content) !== null);
}

function create_text_file(data){
  var output = "---\n";

}


function move_11(content){
  
}

function set_published(content){
  
}

function process_all(){
  // Get all broken files
  // Check 
}

File: admin_scripts/remove_11.rb
# Load all _ssc_projects
# Check each for an "11" key
# If true
#   body = content["11"]
#   content.delete["11"]
#   reconstruct and write_file
# print out some progress

require 'yaml'

ROOT_DIR = "../_ssc_projects"
YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m

def project_filenames
  Dir.entries(ROOT_DIR).reject do |filename|
    %w{. .. .DS_Store}.include? filename
  end
end

def load_content(filename)
  content = File.read(File.join(ROOT_DIR, filename))

  if content =~ YAML_FRONT_MATTER_REGEXP
    body = $'
    data = YAML.load($1)
    {body: body, data: data}
  end
end

def broken_content(content)
  data = content[:data]
  data.keys.include? "11"
end

def fix_content(content)
  body = content[:data]["11"]
  content[:data].delete "11"
  data = content[:data]
  {body: body, data: data}
end

def write_file(filename, content)
  content = fix_content(content)
  File.open('output.txt', 'w') do |file|
    file << "---"
    file << YAML.dump(content[:data])
    file << "---\n"
    file << content[:body]
  end
end

def process_all
  # get all files
  # for each
  project_filenames.each do |filename|
    content = load_content(filename)
    # check if broken
    if broken_content(content)
      # puts "Need to fix #{filename}"
      fixed_content = fix_content(content)
      puts "Fixed #{filename}", fixed_content
    end
  end
  #   if broken, fix
  #   write
  #   log/mark progress
end

# data = load_content("00045776.txt")[:data]
# content = load_content("xxx742ba697.txt")
# data = content[:data]
# puts broken_content(content)
# write_file(nil, content)
process_all